servers/src/main/java/tachyon/web/WebInterfaceWorkersServlet.java

```diff
@@ -126,19 +132,20 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response)
    * @param workerInfos The list of WorkerInfo objects
    * @return The list of NodeInfo objects
    */
-  private NodeInfo[] generateOrderedNodeInfos(List<WorkerInfo> workerInfos) {
+  private NodeInfo[] generateOrderedNodeInfos(Collection<WorkerInfo> workerInfos) {
     NodeInfo[] ret = new NodeInfo[workerInfos.size()];
-    Collections.sort(workerInfos, new Ordering<WorkerInfo>() {
-      @Override
-      public int compare(WorkerInfo info0, WorkerInfo info1) {
-        return info0.getAddress().getHost().compareTo(info1.getAddress().getHost());
-      }
-    });
     int index = 0;
     for (WorkerInfo workerInfo : workerInfos) {
       ret[index ++] = new NodeInfo(workerInfo);
     }
 
+    Arrays.sort(ret, new Ordering<NodeInfo>() {
+      @Override
+      public int compare(NodeInfo info0, NodeInfo info1) {
+        return info0.getHost().compareTo(info1.getHost());
+      }
+    });
```

Comment:
That's a good idea. I also made NodeInfo private, since it was used nowhere else.